hal_st/middlewares/ble_middleware/GapSt.cpp

```diff
@@ -32,6 +32,11 @@ namespace hal
                         return services::GapPairingObserver::PairingErrorType::unknown;
                 }
         }
+
+        constexpr uint8_t RfWakeupClockSelection(const hal::GapSt::RfWakeupClock& rfWakeupClock)
+        {
+            return static_cast<uint8_t>(rfWakeupClock == hal::GapSt::RfWakeupClock::highSpeedExternal ? SHCI_C2_BLE_INIT_CFG_BLE_LS_CLK_HSE_1024 : SHCI_C2_BLE_INIT_CFG_BLE_LS_CLK_LSE);
```

Comment:
If my other comment was read, then here it could change to: 

```suggestion
        constexpr auto RfWakeupClockSelection(const hal::GapSt::RfWakeupClock& rfWakeupClock)
        {
            return infra::enum_cast(rfWakeupClock);
```